### ISSUE TITLE
Tasking Manager does not load

### ISSUE THREAD

**Issue by OttoROSM** [hotosm/tasking-manager#886]
When browsing to: _http://tasks.hotosm.org/_ the only thing that loads is the HOT Tasking Manager logo.
This happens on my laptop using several browsers:
- Chrome
- FireFox
- Internet Explorer
- Edge

I'm running the latest Java and Windows 10 fall update.
Via Chrome om IOS I have no issue in accessing the tasking manager.

**Comment by OttoROSM** [comment]
I found out what is causing it. I am using Heimdal pro and this site blocked traffic to cdn.jsdelivr.net
After I switched Heimdal off, reloaded HOT and turned it on again, the site loads fine again.

**Comment by ethan-nelson** [comment]
Do we have any interest in hosting the JS resources ourselves?

**Comment by pgiraud** [comment]
@ethan-nelson I was also suprised recently to figure out that we are not hosting the resources ourselves. Though I understand the interest to use CDN's it prevented me from working on some development while I was offline.

**Comment by verdy-p** [comment]
The problem is not the use of any CDN, but the current choice of CDN used, which is used by too many rogue spammers (most often for very intrusive ads fill of privacy issues in their scripts).

If you have nothing better, consider hosting the script elsewhere I even think that it would be fine to host these scripts in a Wikimedia user page or Wikimedia Labs host: at least there, the hosting envioronment has strong poilicies about respecting user privacy and security, and there's little chance it will be blocked. However on an Wikipedia user page, there's the risk of being blocked in some countries that don't like contents hosted on Wikipedia (notably being blocked in China by their national infrastructure based on mandatory "transparent proxies" applied by local ISPs).

There's much less risks being blocked by third parties (notably antispam/animalwares and other security tools) if they are hosted the Wikimedia Labs farm (but it requires an active project to be maintained there and some presence to reply to incoming issues, otherwise the Wikimedia Labs-hosted project may be closed administratively: make sure you correctly describe the project, and have suitable licences and you keep it open, there's no reason Wikimedia will block scripts used by OSM task managers when we have good cooperation with them). Wikimedia is very active at fighting abuses and solving any issue that could happen if one hosted project is harvested by malicious third parties, spamwares, spywares, virus hosting, and so on: it can clean up these projects rapidely and make them live again very fast, thanks to its very active local community of maintainers and admins.

If you have limited resources on your servers for HOT TM, you could also request some hosting to the OSM Foundation or one of its partners (including local chapters that have spare servers, notably OSM France and OSM Germany, with resources sometimes also granted or financed by their own local partners such as local foundations, donations from ISPs, universities, national and regional public services, or some interested organizations that will "pay the bill" for years without requiring to host these contents themselves under their own end-user-agreements and other requirements such as tracking or advertizing or limitations by volume).

**Comment by verdy-p** [comment]
note that the "jsDeliver" CDN has been too frequently used rogue spammers and for illegal activities
Just one example: https://hackerone.com/jsdelivr

"jsDeliver" is too long to react to incoming reports of abuses, and probably does not have enough admins with the necessary technical skills to investigate. That's why it may be blocked by several security suites when the rate of abuses is constantly growing and no actions is taken for too long when thes combined abuses reach a severe level of global attacks.

But may be if you use "jsDeliver" for delivery with HTTPS, we'll not suffer from such blocking (consider making TM's server accessible with HTTPS too, and implementing OAuth or Google authentication for TM registered users, enabled with 2-factor-authentication to secure our own user accounts on the TM and with requests forwarded to OSM API).

**Comment by ethan-nelson** [comment]
Yes I agree about vulnerabilities for hosted elsewhere--someone taking over the domain and injecting (e.g. [[1]](https://blog.malwarebytes.com/threat-analysis/2017/10/equifax-transunion-websites-push-fake-flash-player/)). I also see the upside related to caching if multiple sites use the same CDN.

@david-hotosm do you have any input about hosting JS resources through our resources instead of using an external site?

**Comment by ethan-nelson** [comment]
Closing for now since original issue is closed--moving JS hosting to new ticket.
